### PR TITLE
Update training notebooks 02, 03, 05 and 06

### LINKED ISSUE #70: Re-run and update exercise 02, 03, 05 and 06 for Oct 2023 training session
Gibbs notebook should match paper I.

- [x] Exercise 02
- [x] Exercise 03
- [x] Exercise 05
- [x] Exercise 06

### PR DESCRIPTION
**Pull request by nabriis**
Closes #70.

Did minor updates to ensure notebooks run etc.

### CODE REVIEW

File: training/Exercise02_Distributions.ipynb [hunk 1]
```diff
@@ -318,28 +318,26 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## 2. Multivariate distributions <a class=\"anchor\" id=\"Multivariate\"></a> "
+    "## 2. Multivariate distributions <a class=\"anchor\" id=\"Multivariate\"></a> \n",
+    "\n",
+    "CUQIpy currently implements a number multivariate distributions in the `cuqi.distribution` module:"
```
Comment (amal-ghamdi) [inline training/Exercise02_Distributions.ipynb:323]:
```suggestion
    "CUQIpy currently implements a number of multivariate distributions in the `cuqi.distribution` module:"
```

File: training/Exercise03_Forward_Models.ipynb [hunk 2]
```diff
@@ -565,7 +565,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Some LinearModel objects simply contain a matrix representing the linear mapping; for those this matrix can be accessed by:"
+    "Some LinearModel objects simply contain a matrix representing the linear mapping; while others use a matrix-free approach. In both cases, one can extract the matrix of the linear model by (this will raise an error if the model is too large):"
```
Comment (amal-ghamdi) [inline training/Exercise03_Forward_Models.ipynb:568]: This is probably to keep in mind but not part of this PR, it could be a good idea to inform the user that a matrix being built when this is the case. i.e. when asking for matrix and the linear model is matrix-free. Do we do that?

File: training/Exercise03_Forward_Models.ipynb [hunk 3]
```diff
@@ -795,7 +795,9 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Note that the mean of the Gaussian is really `A*x`, but since `x` is the forward parameter of `A`, this is implied. Work is underway on CUQIpy to enable specifying diretly `A*x` as the mean, but currently this will produce an error that `x` is undefined. A hack to make it work at present is to specify `x` as any distribution of the size of A's domain range and use that as a placeholder. In this way the name of the conditioning variable can also be changed, for example here we use `u` instead of `x`"
+    "Note that the mean of the Gaussian is really `A*x`, but since `x` is the forward parameter of `A`, this is implied. \n",
+    "\n",
+    "When working multiple distributions, one can be more explicit about the parameter to the forward model by explicitly `evaluating` the forward model with the desired distribution as input. In this way the forward parameter can also be changed from e.g. the default `x`` to `u`:"
```
Comment (amal-ghamdi) [inline training/Exercise03_Forward_Models.ipynb:800]:
```suggestion
    "When working multiple distributions, one can be more explicit about the parameter to the forward model by explicitly `evaluating` the forward model with the desired distribution as input. In this way the forward parameter can also be changed from e.g. the default `x` to `u`:"
```

File: training/Exercise06_Gibbs.ipynb [hunk 4]
```diff
@@ -716,16 +718,16 @@
...
+    "One choice is the [LMRF](https://cuqi-dtu.github.io/CUQIpy/api/_autosummary/cuqi.distribution/cuqi.distribution.LMRF.html) prior, which assumes a Laplace distribution for the differences between neighboring elements of $\\mathbf{x}$ instead of a Gaussian. That is we assume\n",
     "\n",
     "\n",
     "\\begin{align*}\n",
-    "\\mathbf{x} \\sim \\text{Laplace-diff}(\\mathbf{0}, d^{-1}),\n",
+    "\\mathbf{x} \\sim \\text{LMRF}(\\mathbf{0}, d^{-1}),\n",
     "\\end{align*}\n",
     "\n",
-    "which translates to the assumption that $x_i-x_{i-1} \\sim \\mathrm{Laplace}(0, d^{-1})$ for $i=1, \\ldots, n$.\n",
+    "which translates to the assumption that $x_i-x_{i-1} \\sim \\mathrm{Laplace}(0, d^{-1})$ for $i=1, \\ldots, n$ (excluding boundary conditions)\n",
```
Comment (amal-ghamdi) [inline training/Exercise06_Gibbs.ipynb:730]: not sure?
```suggestion
    "which translates to the assumption that $x_i-x_{i-1} \\sim \\mathrm{Laplace}(0, d^{-1})$ for $i=1, \\ldots, n$ (excluding boundaries)\n",
```

**Review by amal-ghamdi — COMMENTED** [review]
I took a look at the changes, LGTM! I made few suggestions. One point, would it be possible to address issue #74 as part of this PR as well?

Thank you

**Comment by jakobsj** [comment]
Did you mean to request my review on this one? If only very minor changes, I'm happy if @amal-ghamdi is :)

File: training/Exercise03_Forward_Models.ipynb (hunk 2 above)
Comment (nabriis) [reply training/Exercise03_Forward_Models.ipynb:568]: We don't at this point. Unsure if we want to.

**Review by amal-ghamdi — APPROVED** [review]
LGTM

File: training/Exercise03_Forward_Models.ipynb (hunk 2 above)
Comment (amal-ghamdi) [reply training/Exercise03_Forward_Models.ipynb:568]: I would say if it is cached then probably it might be okay not to inform the user, but if it is not cached and computed every time the user asks for it then it will be helpful to update the user.

Comment (nabriis) [reply training/Exercise03_Forward_Models.ipynb:568]: Just checked, and we do cache it :)
